semantic-python/src/Language/Python/Core.hs

```diff
@@ -13,26 +15,35 @@ import           Data.Core as Core
 import           Data.Foldable
 import           Data.Name as Name
 import           GHC.Generics
+import           GHC.Records
+import qualified Data.Loc
 import qualified TreeSitter.Python.AST as Py
 import           TreeSitter.Span (Span)
 
+-- We don't want to commit to a particular representation of Core syntax,
+-- but there are commonalities that repeatedly crop up and that clog type
+-- signatures, so here's a constraint-kind alias for its capabilities.
```

Comment:
I dunno that this is true. `Term (Ann :+: Core)` is fine, and the only thing we gain from abstracting is an assurance that you can’t pattern match on or otherwise eliminate a compiled term.

If that’s the actual goal we have in mind here, we should say that explicitly; otherwise, let’s simplify and say `Term (Ann :+: Core)` instead of `t`.

Reply:
I like that assurance, so I’ll change the wording here.